### PR TITLE
Toward trace validating raft_scenarios/election_while_reconfiguration

### PR DESCRIPTION
**Pull request by lemmy**
Correct mapping between a raft.h configuration and a ccfraft.tla configuration.

Part of https://github.com/microsoft/CCF/issues/5057

### CODE REVIEW

File: tla/Traceccfraft.tla [hunk 1]
```diff
@@ -113,7 +113,7 @@ OneMoreMessage(msg) ==
 VARIABLE l, ts
 
 TraceInit ==
-    /\ l = 1
+    /\ l = 2
```
Comment (achamayou) [inline tla/Traceccfraft.tla:116]: why 2 now?

Comment (lemmy) [reply tla/Traceccfraft.tla:116]: The first line of the log maps to the initial state.  The second line (`l=2`) maps to the second state.  In other words, there is always a stuttering step after the initial state with `l=1`.

File: tla/Traceccfraft.tla [hunk 2]
```diff
@@ -218,7 +218,7 @@ IsChangeConfiguration ==
     /\ IsEvent("add_configuration")
     /\ state[logline.msg.state.node_id] = Leader
     /\ LET i == logline.msg.state.node_id
-           newConfiguration == logline.msg.new_configuration
+           newConfiguration == DOMAIN logline.msg.new_configuration.nodes
```
[the remark this answers is not in the document]
Comment (lemmy) [reply tla/Traceccfraft.tla:221]: We log `configurations` and `new_configuration`, i.e., the former is a set (array) of configurations while the latter is a single config:

```json
"new_configuration":{"idx":2,"nodes":{"0":{"address":":"},"1":{"address":":"},"2":{"address":":"},"3":{"address":":"},"4":{"address":":"}},"rid":2}
```

```json
"configurations":[{"idx":0,"nodes":{"0":{"address":":"},"1":{"address":":"},"2":{"address":":"},"3":{"address":":"},"4":{"address":":"}},"rid":0}]
```

The actions `IsChangeConfiguration` and `ChangeConfigurationInt` take as parameter `new_configuration` and not ccfraft's variable `configurations`.

[the remark this answers is not in the document]
Comment (lemmy) [reply tla/Traceccfraft.tla]: Corrected/aligned to `ToConfigurations(<<logline.msg.new_configuration>>)`

**Comment by lemmy** [comment]
The newest commits validate `election_while_reconfiguration` up to length 6, and `reconfiguration` up to length 18.  However, if the conjunct `/\ logline.msg.sent_idx + 1 = nextIndex[i][j]` is removed from `IsSendAppendEntries`, `election_while_reconfiguration` validates up to 11 and `reconfiguration` up to 48.

**Review by achamayou — APPROVED** [review]

**Pull request merged**